### PR TITLE
Move functions defined in constants.py in config.py

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY

Move functions defined in constants.py in config in order to allow other program (for example ansibullbot) to reuse these functions.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
constants.py

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel 2af5556901) last updated 2017/05/26 00:24:32 (GMT +200)
  config file = 
  configured module search path = [u'/home/pilou/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/pilou/ansible/lib/ansible
  executable location = /home/pilou/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```

### CODE REVIEW

**Comment by sivel** [comment]
This change will also impact 3rd party tools that are already importing these functions from constants.

What prevents you from importing things from constants as is?

These change seems unnecessary.

**Comment by pilou-** [comment]
All ansible related variables/constants are useless when importing `get_config_file/mk_boolean/shell_expand` methods from another project.

**Comment by sivel** [comment]
Just because there are extra vars and functions there doesn't mean you have to import them.  Do you have a use case in which the current code organization actually causes a real problem?

The following import works fine without any issues for me.

```
from ansible.constants import get_config, mk_boolean
```

I'm sure in many circumstances when you are importing things from a package, you are never importing them all, but that doesn't mean that you should need to separate only the things you want to use into another file.

File: lib/ansible/constants.py [hunk 1]
```diff
@@ -20,155 +20,19 @@
...
-            try:
-                p.read(path)
-            except configparser.Error as e:
-                raise AnsibleOptionsError("Error reading config file: \n{0}".format(e))
-            return p, path
-    return None, ''
 
+try:
+    p, CONFIG_FILE = load_config_file()
+except configparser.Error as e:
```
Comment (sivel) [inline lib/ansible/constants.py]: Why move the exception handling out of the `load_config_file` function?

File: lib/ansible/executor/playbook_executor.py [hunk 2]
```diff
@@ -192,7 +193,7 @@ def run(self):
                         retries = sorted(retries)
                         if len(retries) > 0:
                             if C.RETRY_FILES_SAVE_PATH:
-                                basedir = C.shell_expand(C.RETRY_FILES_SAVE_PATH)
+                                basedir = shell_expand(C.RETRY_FILES_SAVE_PATH, C.CONFIG_FILE)
```
Comment (sivel) [inline lib/ansible/executor/playbook_executor.py]: I'm also concerned that we are needing to change function signatures, to allow moving the code out, for what seems to be little benefit

**Comment by pilou-** [comment]
> I'm sure in many circumstances when you are importing things from a package, you are never importing them all, but that doesn't mean that you should need to separate only the things you want to use into another file.

When using the following snippet:
`from ansible.constants import mk_boolean`
`get_config` method is called more than 150 times. When reusing `mk_boolean` from another project (for example `ansibullbot`), it seems unnecessary.

File: lib/ansible/constants.py (hunk 1 above)
Comment (pilou-) [reply lib/ansible/constants.py]: Updated: I kept the `load_config_file` in `constants.py`.

**Comment by bcoca** [comment]
I was doing something similar in  #12797 ... it also breaks backwards compat

**Comment by pilou-** [comment]
@bcoca would you advise to improve this pull-request or let it go ?

**Comment by bcoca** [comment]
@pilou- .. not sure ... I'm about to 'refresh' ansible-config as it is on roadmap for 2.4 but have not started yet.

**Comment by abadger** [comment]
@pilou- I think we should close this PR due to the new config work done in devel:

* mk_boolean() is now the boolean function here: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/parsing/convert_bool.py
* get_config has been changed radically: https://github.com/ansible/ansible/blob/devel/lib/ansible/config/  (You have to instantiate a ConfigManager()) and then retrieve the settings from the ConfigData() object that is part of the ConfigManager().

I'm not sure if the replacement of get_config will meet your needs.  You probably need to talk to bcoca in irc about it, though, as he's still actively working on it for 2.4.

**Comment by abadger** [comment]
needs_revision

**Comment by abadger** [comment]
needs_info

**Comment by pilou-** [comment]
It seems that's ok :)

**Pull request closed without merging**